source/mainarrays.f90

```diff
@@ -162,3 +162,148 @@ subroutine expand_thickarrays(nele_request, npart_request, nblz_request, nblo_re
+    naaLast(j:tnapx)   = cshift(naaLast(j:tnapx),   1)
+    nzzLast(j:tnapx)   = cshift(nzzLast(j:tnapx),   1)
+
+    tnapx = tnapx - 1
+  end do
+  napx_new = tnapx
+
+  ! Collimation
+  if(do_coll) then
+    tnapx = napx
+    do j=napx,1,-1
+      if(llostp(j) .eqv. .false.) cycle
+
+      xgrd(j:tnapx)                 = cshift(xgrd(j:tnapx),                 1)
+      ygrd(j:tnapx)                 = cshift(ygrd(j:tnapx),                 1)
+      xpgrd(j:tnapx)                = cshift(xpgrd(j:tnapx),                1)
+      ypgrd(j:tnapx)                = cshift(ypgrd(j:tnapx),                1)
+      pgrd(j:tnapx)                 = cshift(pgrd(j:tnapx),                 1)
+      ejfvgrd(j:tnapx)              = cshift(ejfvgrd(j:tnapx),              1)
+      sigmvgrd(j:tnapx)             = cshift(sigmvgrd(j:tnapx),             1)
+      rvvgrd(j:tnapx)               = cshift(rvvgrd(j:tnapx),               1)
+      dpsvgrd(j:tnapx)              = cshift(dpsvgrd(j:tnapx),              1)
+      oidpsvgrd(j:tnapx)            = cshift(oidpsvgrd(j:tnapx),            1)
+      dpsv1grd(j:tnapx)             = cshift(dpsv1grd(j:tnapx),             1)
+
+      part_hit_pos(j:tnapx)         = cshift(part_hit_pos(j:tnapx),         1)
+      part_hit_turn(j:tnapx)        = cshift(part_hit_turn(j:tnapx),        1)
+      part_abs_pos(j:tnapx)         = cshift(part_abs_pos(j:tnapx),         1)
+      part_abs_turn(j:tnapx)        = cshift(part_abs_turn(j:tnapx),        1)
+      part_select(j:tnapx)          = cshift(part_select(j:tnapx),          1)
+      part_impact(j:tnapx)          = cshift(part_impact(j:tnapx),          1)
+      part_indiv(j:tnapx)           = cshift(part_indiv(j:tnapx),           1)
+      part_linteract(j:tnapx)       = cshift(part_linteract(j:tnapx),       1)
+      part_hit_before_pos(j:tnapx)  = cshift(part_hit_before_pos(j:tnapx),  1)
+      part_hit_before_turn(j:tnapx) = cshift(part_hit_before_turn(j:tnapx), 1)
+
+      secondary(j:tnapx)            = cshift(secondary(j:tnapx),            1)
+      tertiary(j:tnapx)             = cshift(tertiary(j:tnapx),             1)
+      other(j:tnapx)                = cshift(other(j:tnapx),                1)
+      scatterhit(j:tnapx)           = cshift(scatterhit(j:tnapx),           1)
+      nabs_type(j:tnapx)            = cshift(nabs_type(j:tnapx),            1)
+      ipart(j:tnapx)                = cshift(ipart(j:tnapx),                1)
+      flukaname(j:tnapx)            = cshift(flukaname(j:tnapx),            1)
+
+      counted_r(j:tnapx,:)          = cshift(counted_r(j:tnapx,:),          1, 1)
+      counted_x(j:tnapx,:)          = cshift(counted_x(j:tnapx,:),          1, 1)
+      counted_y(j:tnapx,:)          = cshift(counted_y(j:tnapx,:),          1, 1)
+
+      tnapx = tnapx - 1
+    end do
+  end if
+
+#ifdef FLUKA
+  if(fluka_enable) then
+    tnapx = napx
+    do j=napx,1,-1
+      if(llostp(j)) call fluka_lostpart(tnapx, j) ! Inform fluka
```

Comment:
we have to apply the same logics in `fluka_lostpart`

Reply:
I looked at fluka_lostpart, and it looked to me that this was equivalent, but it doesn't retain the old information. I assumed that was not necessary because it is anyway overwritten.

However, I'm happy to rewrite the routine to do the same thing.

Reply:
Also spotted a bug in that loop. `tnapx = tnapx - 1` should be inside the if statement.